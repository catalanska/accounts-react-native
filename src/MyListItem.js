import React from 'react';
import PropTypes from 'prop-types';
import { Text, View, Image, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  image: {
    borderRadius: 5,
    height: 70,
    width: 70,
    resizeMode: 'contain',
    margin: 0,
  },
  row: {
    height: 70,
    flex: 1,
    flexDirection: 'row',
  },
});

const MyListItem = ({ image, title, account }) => (
  <View style={styles.row}>
      <Image
        style={styles.image}
        source={image}
      />
      <Text style={{backgroundColor: 'steelblue'}}>
        {title}
      </Text>
      <Text style={{backgroundColor: 'red'}}>
        {account}
      </Text>
    </View>
);

MyListItem.propTypes = {
  image: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  account: PropTypes.string.isRequired,
};

export default MyListItem;
