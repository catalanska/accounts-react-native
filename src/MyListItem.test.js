// __tests__/Intro-test.js
import 'react-native';
import React from 'react';
import MyListItem from './MyListItem';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('MyListItem should', () => {
  test('render correctly when no params', () => {
    const rendered = renderer.create(<MyListItem />).toJSON();

    expect(rendered).toMatchSnapshot();
  });

  test('render correctly when params are passed', () => {
    const item = <MyListItem id={1} title='test' image={1} account='f00' />;

    const rendered = renderer.create(item).toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
