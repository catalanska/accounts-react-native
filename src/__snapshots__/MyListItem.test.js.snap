// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyListItem should render correctly when no params 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "height": 70,
    }
  }
>
  <Image
    source={undefined}
    style={
      Object {
        "borderRadius": 5,
        "height": 70,
        "margin": 0,
        "resizeMode": "contain",
        "width": 70,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "steelblue",
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "red",
      }
    }
  />
</View>
`;

exports[`MyListItem should render correctly when params are passed 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "height": 70,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "borderRadius": 5,
        "height": 70,
        "margin": 0,
        "resizeMode": "contain",
        "width": 70,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "steelblue",
      }
    }
  >
    test
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "red",
      }
    }
  >
    f00
  </Text>
</View>
`;
